RATapi/classlist.py

```diff
@@ -296,11 +296,15 @@ def extend(self, other: Sequence[T]) -> None:
         self._check_unique_name_fields(other)
         self.data.extend(other)
 
-    def set_fields(self, index: int, **kwargs) -> None:
+    def set_fields(self, index: Union[int, slice, str, T], **kwargs) -> None:
```

Comment:
I think this should be `Union[int, slice, str]`?

Reply:
Should I add the object case to the tests maybe?

Reply:
Done!